pynamodb/attributes.py

```diff
@@ -253,7 +253,10 @@ def append(self, other: Iterable) -> '_ListAppend':
     def prepend(self, other: Iterable) -> '_ListAppend':
         return Path(self).prepend(other)
 
-    def set(self, value: Any) -> 'SetAction':
+    def set(
```

Comment:
Let's add some negative tests to typing_tests, i.e. where you pass stuff that shouldn't work, and have to `# type: ignore[some-code]` it.

We're using the [warn_unused_ignores technique](https://typing.readthedocs.io/en/latest/source/quality.html#testing-using-assert-type-and-warn-unused-ignores) here: adding a `# type: ignore` will only pass mypy check if it's actually suppressing what otherwise would be an error.